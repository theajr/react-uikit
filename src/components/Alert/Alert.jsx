import React from 'react'
import PropTypes from 'prop-types'
import cx from '../../utils/cx'

export const alertTypes = ['primary', 'success', 'warning', 'danger']

const Alert = ({ type, closable, animation, duration, children }) => {
  const ukAlert = ''
  const classes = cx({
    'uk-alert': true,
    [`uk-alert-${type}`]: type
  })
  return (
    <div uk-alert={ukAlert} className={classes}>
      {closable && <a className='uk-alert-close' uk-close='' />}
      {children}
    </div>
  )
}

Alert.propTypes = {
  type: PropTypes.oneOf(alertTypes),
  closable: PropTypes.bool,
  animation: PropTypes.bool,
  duration: PropTypes.number,
  children: PropTypes.node
}

export default Alert
