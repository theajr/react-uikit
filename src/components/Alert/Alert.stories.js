import React from 'react'
import { withKnobs, text, boolean, radios } from '@storybook/addon-knobs'
import Alert, { alertTypes } from './Alert'
import { withInfo } from '@storybook/addon-info'
import { titleCase } from '../../utils/str'

export default {
  title: 'Alert',
  decorators: [withKnobs, withInfo({ source: true })]
}

export const Gallery = () => {
  const closable = boolean('Closable?', false)

  const type = radios(
    'type',
    alertTypes.reduce((acc, i) => ({ ...acc, [titleCase(i)]: i }), {})
  )

  return (
    <Alert closable={closable} type={type}>
      {text('Text', 'Hello, World')}
    </Alert>
  )
}
