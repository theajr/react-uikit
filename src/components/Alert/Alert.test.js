import React from 'react';
import { shallow } from 'enzyme';
import Alert from './Alert';

describe('<Alert />', () => {
  test('renders', () => {
    const wrapper = shallow(<Alert />);
    expect(wrapper).toMatchSnapshot();
  });
});
