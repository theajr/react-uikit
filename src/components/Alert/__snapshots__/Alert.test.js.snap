// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> renders 1`] = `ShallowWrapper {}`;
